lib/mongoid/matchable/default.rb

```diff
@@ -69,6 +69,32 @@ def determine(value, operator)
           attr ? attr.send(operator, first(value)) : false
         }
       end
+
+      # Convenience method for checking _matches? on a Document or a Hash.
+      #
+      # @example Does the document of type Document or Hash match?
+      #   safe_matches?(default, "a", 1)
+      #   safe_matches?(:a => 1, "b", 2)
+      #
+      # @param [ Document, Hash ] document The object of type Document or Hash to call _matches? on
+      # @param [ String ] key The key
+      # @param [ Object ] value The value to check if it matches.
+      #
+      # @return [ true, false ] True if matches, false if not.
+      #
+      # @since 7.0.4
+      def safe_matches?(document, key, value)
+        if document.is_a?(Document)
+          document._matches?(key => value)
+        else
```

Comment:
`elsif document.is_a?(Hash) && document.present?`

They you can get rid of all the `try` stuff

Reply:
Good idea. What should the behavior be if this new "if" branch isn't taken? No-op, log, error?